### PR TITLE
Update in-code documentation for util/Version.h and related cpp files

### PR DESCRIPTION
**Pull request by adrianbroher**
Use Doxygen keywords were applicable or required to create proper docs.

Gave it already an aspell check for spelling errors.

### CODE REVIEW

**Comment by geoffthemedio** [comment]
It's just some comments... Did you want some feedback before merging, or is there some other reason to make this a pull request first?

**Comment by adrianbroher** [comment]
> Did you want some feedback before merging, or is there some other reason to make this a pull request first?

Yes, I'm not sure if the phrasing inside the documentation is proper.

> It's just some comments

If those kind of PRs are not welcome I can also commit to master without PRs.

**Comment by geoffthemedio** [comment]
You're certainly welcome to request review of any code before committing it to master.

File: util/Version.cpp.in [hunk 1]
```diff
@@ -1,5 +1,13 @@
 #include "util/Version.h"
 
+/** @file
+ * @brief  Implement a free function to access the application version.
+ *
+ * @warning  DON'T EDIT MANUALLY Version.cpp.  @e Version.cpp is generated from
+ *      @e Version.cpp.in by the @e cmake/make_versioncpp.py utility script
+ *      used by all build systems.
```
Comment (geoffthemedio) [inline util/Version.cpp.in]: might say "is or should be used by all..."

File: util/Version.h [hunk 2]
```diff
@@ -1,15 +1,41 @@
 #ifndef _Version_h_
 #define _Version_h_
 
+/** @file
+ * @brief  Declares free functions to access the application and dependencies
```
Comment (geoffthemedio) [inline util/Version.h]: "dependency versions" or "dependencies' versions". I'd prefer the former.

File: util/Version.h [hunk 3]
```diff
@@ -1,15 +1,41 @@
...
+
 #include <string>
 #include <map>
 
 #include "Export.h"
 
+/** @brief  Returns the version string of FreeOrion.
+ *
+ * @return  The version string with the format
+ *      <em>VERSION BRANCH_NAME [build DATE.SHORT_COMMIT_HASH] BUILD_SYSTEM</em>
```
Comment (geoffthemedio) [inline util/Version.h:28]: might want to give an example of what "VERSION" would look like...
might want to clarify that it's a git commit hash.
might want to give example(s) of what BUILD_SYSTEM would look like
or, could just give an example of a full version string.

File: util/Version.h [hunk 4]
```diff
@@ -1,15 +1,41 @@
...
+ */
 FO_COMMON_API const std::string& FreeOrionVersionString();
 
+/** @brief  Returns a map of dependency versions.
+ *
+ * @return  A map with the dependency versions.  The key represents the
+ *      dependency name, the value corresponding dependency version string.
+ *
+ * @bug  The current implementation doesn't expose all dependencies of all
```
Comment (geoffthemedio) [inline util/Version.h]: Is it really a bug? Depends if a "limitation" is a bug... Or could say it just returns the dependency versions for common code.

File: util/Version.h [hunk 5]
```diff
@@ -1,15 +1,41 @@
...
+ *      application components.  For example the human client has a dependency
+ *      on Vorbis but because this function is part of common it isn't added to
+ *      the dependency map.
+ */
 FO_COMMON_API std::map<std::string, std::string> DependencyVersions();
 
+/** @brief  Log the map returned by DependencyVersions() into the @e debug log
+ *      channel.
+ *
+ * @bug  The @e debug log channel isn't the right place to log this but we don't
```
Comment (geoffthemedio) [inline util/Version.h]: arguably unnecessary editorializing in this context...

**Review by geoffthemedio — CHANGES_REQUESTED** [review]

File: util/Version.h (hunk 4 above)
Comment (adrianbroher) [reply util/Version.h]: I would consider it a bug, given the current indirect usage of this function by HumanClientApp::HumanClientApp() and ServerApp::ServerApp() and the in-code comment written by you regarding Vorbis and Ogg.

File: util/Version.h (hunk 5 above)
Comment (adrianbroher) [reply util/Version.h]: Well, okay.  I can remove the bug note but I think the statement in itself holds merit.

Comment (geoffthemedio) [reply util/Version.h]: It is a reasonable use case for an info channel.

File: util/Version.h (hunk 4 above)
Comment (geoffthemedio) [reply util/Version.h]: I suppose you could rename this to CommonDependencyVersions and have client/server-specific equivalents as well.

**Comment by adrianbroher** [comment]
Updated comments as requested.

Split up the implementation of DependencyVersions to each executable dependencies.
@Vezzra the Xcode project needs to be updated.  `util/DependencyVersions.cpp` needs to removed from the `common` project and added to `freeorion`, `freeorionca` and `freeoriond`.

Also added and `info` and `warn` channel and changedthe dependency version logging from the `debug` to the `info` channel.

**Comment by Vezzra** [comment]
Ok, updated the Xcode project.

However, I can't build this branch, getting the following compile errors (with and without the changes to the Xcode project):

```
/Users/user/SoftwareProjekte/FO/src/util/DependencyVersions.cpp:51:30: error: use of undeclared identifier 'vorbis_version_string'; did you mean 'VorbisVersionString'?
        const char* retval = vorbis_version_string();
                             ^~~~~~~~~~~~~~~~~~~~~
                             VorbisVersionString
/Users/user/SoftwareProjekte/FO/src/util/DependencyVersions.cpp:50:17: note: 'VorbisVersionString' declared here
    std::string VorbisVersionString() {
                ^
/Users/user/SoftwareProjekte/FO/src/util/DependencyVersions.cpp:51:21: error: no viable conversion from 'std::string' (aka 'basic_string<char>') to 'const char *'
        const char* retval = vorbis_version_string();
                    ^        ~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by geoffthemedio — APPROVED** [review]

**Comment by adrianbroher** [comment]
> However, I can't build this branch

I included the vorbis header with quotes instead of angle brackets.

**Comment by geoffthemedio** [comment]
I'm still getting compile errors from the #include <zlib.h> line for all three projects (human, ai, server)

**Comment by geoffthemedio** [comment]
Same issue with png.h

**Comment by geoffthemedio** [comment]
And ft2build.h

**Comment by geoffthemedio** [comment]
And after adding the freetype directory to the FreeOrion project, I get a link error:

```
2>Link:
2>     Creating library ../../FreeOrion.lib and object ../../FreeOrion.exp
2>DependencyVersions.obj : error LNK2001: unresolved external symbol _vorbis_version_string
```

**Pull request merged**